### ISSUE TITLE
NestJS package.json is not having dependencies after upgrading to 15.9.2

### ISSUE THREAD

**Issue by nosaku** [nrwl/nx#16365]
### Current Behavior

NestJS package.json is not having all the dependencies after upgrading to 15.9.2. In the previous version, 15.8.1 it was working. Seems this bug again came up.

### Expected Behavior

package.json is generated with all the dependencies for the NestJS application.

### Steps to Reproduce

1. Run the NestJS build and see the package.json in the dist folder. It has only the "tslib" dependency and all other dependencies are missing.

### Nx Report

```shell
NA
```

**Comment by nosaku** [comment]
I do like this in my Azure pipelines:

1. npm install --no-save
2. npm install --package-lock
3. build the nestjs app
4. In the dist folder, package.json contains only "tslib" as dependency and all other dependencies used by app are missing.

**Comment by nosaku** [comment]
Happens only in Azure pipelines.. locally same steps does generate the dependencies correctly.

**Comment by alumni** [comment]
I have this problem in 16.0.0.beta.1.

It was fine for me in 15.9.2 and 16.0.0.beta.0, but it is no longer fine in 16.0.0.beta.1.

**Comment by nosaku** [comment]
For me locally in Windows 10 it works fine with 15.9.2. But when I run from Azure pipeline in ubuntu-latest agent it doesn't generate the dependencies. Very strange.

**Comment by nosaku** [comment]
Just tried with 15.8.5 through 15.8.9 and it works... starting from 15.9.0 there is the problem.

**Comment by mandarini** [comment]
Possibly related? https://github.com/nrwl/nx/issues/16320

**Comment by fguitton** [comment]
This definitely expends beyond `@nrwl/nest`, I am seeing the same thing on `@nrwl/node` and `@nrwl/wepback`, as well from `15.9.0` onward. Likely some shared piece of configuration indeed.

**Comment by SolarLiner** [comment]
I've been hit with this using `@nrwl/webpack:webpack` on 16.0.0.beta.5 as well. I fixed it by recreating the lockfile with the latest version of pnpm -- in particular, check that the `lockfileVersion` is set to `'6.0'` after recreating the lockfile. It makes me think that this error comes from a parsing mismatch between versions of the lockfile.

**Comment by alumni** [comment]
I'm using yarn v1 in this project :)

**Comment by fguitton** [comment]
@nosaku @alumni I've done a bit of digging, the problem starts showing up between `15.9.0-beta.0` and `15.9.0-beta.1`.
Looking at the Nx codebase for a bit, I saw some behaviour changes around how the `inputs` from `project.json`. If you have a extra `fileset` for example all the default project dependencies are ignored.

Internally, in the `create-json-package.ts` file, Nx goes from inferring :
```
selfInputs === [
  '{projectRoot}/**/*',
  '{workspaceRoot}/babel.config.json',
  '!{projectRoot}/**/?(*.)+(spec|test).[jt]s?(x)?(.snap)',
  '!{projectRoot}/tsconfig.spec.json',
  '!{projectRoot}/jest.config.[jt]s',
  '!{projectRoot}/.eslintrc.json'
]
dependencyInputs === [
  '{projectRoot}/**/*',
  '{workspaceRoot}/babel.config.json',
  '!{projectRoot}/**/?(*.)+(spec|test).[jt]s?(x)?(.snap)',
  '!{projectRoot}/tsconfig.spec.json',
  '!{projectRoot}/jest.config.[jt]s',
  '!{projectRoot}/.eslintrc.json'
]
```
To inferring thigs :
```
selfInputs === [ '{workspaceRoot}/node_modules/.prisma/**/*.ts' ]
dependencyInputs === []
```

The problem is still upstream of that as it has to do with the populating of the `target` metadata.
Perhaps if you try to add the defaults explicitly to your `inputs` field your problem would lift ?

Best wishes,

**Comment by nosaku** [comment]
@fguitton  Thank you for the detailed response. I will give a try. What should I include in "inputs"? I don't use this in current project.json file.

**Comment by nosaku** [comment]
Just added this and tried but didn't work.

"inputs": [
        "production"
      ],

**Comment by nosaku** [comment]
Hi @fguitton I am afraid I can't share the entire project.json file, but I guess this is the snipped which may help. Please let me know if you need any other sections too of the file.

      "executor": "@nrwl/webpack:webpack",
      "outputs": [
        "{options.outputPath}"
      ],
      "inputs": [
        "production"
      ],
      "options": {
        "generatePackageJson": true,
        "outputPath": "dist/apps/xxx",
        "main": "apps/xxx/src/main.ts",
        "tsConfig": "apps/xxx/tsconfig.app.json",
        "assets": [
          "apps/xxx/src/assets"
        ],
        "target": "node",
        "compiler": "tsc"
      },

**Comment by nosaku** [comment]
![image](https://user-images.githubusercontent.com/5987370/234876569-619dd2e0-a9a3-4b9b-ac45-f00e097222cb.png)

is this causing issue?

**Comment by alumni** [comment]
@fguitton The error that I had was introduced between 16.0.0-beta.0 and 16.0.0-beta.1 and was fixed in 16.0.0.
